#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add('binary_threshold', int_t, 0, 'Greyscale threshold for binarisation of image in order to find reference marker contours', 140, 0, 255)
gen.add('centrey_min', double_t, 0, 'Minimum reference marker centre y-coordinate', 170, 0, 500)
gen.add('radius_max', double_t, 0, 'Maximum reference marker radius', 18, 0, 50)
gen.add('aspect_max', double_t, 0, 'Maximum reference marker aspect ratio', 3, 1, 5)
gen.add('area_min', double_t, 0, 'Minimum reference marker area', 45, 0, 200)
gen.add('area_max', double_t, 0, 'Maximum reference marker area', 500, 300, 800)

gen.add('track_tol', double_t, 0, 'Reference marker tracking tolerance', 55, 0, 200)
gen.add('track_age_min', int_t, 0, 'Minimum track age in order to be considered', 10, 1, 60)
gen.add('debounce_dist', double_t, 0, 'Minimum change in centre in order to be recorded to CSV again', 20, 0, 100)
gen.add('smooth_ts', double_t, 0, 'Settling time of smoothed reference marker detections (used for stability determination)', 0.5, 0.01, 2)
gen.add('stable_dist', double_t, 0, 'Maximum distance to smoothed reference marker to classify as stable', 2, 0, 10)

exit(gen.generate(pkgname='nicol_demos', nodename='camcalib_data_collector', name='CamCalibDataCollector'))
# EOF
